Incompatible with OpenBSD

The Makefile is not compatible with OpenBSD.
After I set ${CC} to `cc` in the Makefile, `gmake` (GNU make) gives the following error:

```
cc -o xsnip xsnip.c -g -I. -lX11 -lpng
xsnip.c:26:10: fatal error: 'png.h' file not found
#include <png.h>
         ^~~~~~~
1 error generated.
gmake: *** [Makefile:14: xsnip] Error 1
```

png.h on OpenBSD is located at /usr/local/include/png.h.
Projects like dwm achieve cross-platform compatibility by including a few commented out lines that once uncommented, will make make (or cc) known to the real location of the libraries.

I might try to fix this issue myself and submit a PR.

--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,14 @@
-CC=gcc
-CFLAGS=-g -I. -lX11 -lpng
+X11_CFLAGS != pkg-config --cflags x11
+X11_LIBS != pkg-config --libs x11
+
+LIBPNG_CFLAGS != pkg-config --cflags libpng
+LIBPNG_LIBS != pkg-config --libs libpng
+
+CC?=gcc
+CFLAGS=-g ${X11_CFLAGS} ${LIBPNG_CFLAGS}
+LIBS=${X11_LIBS} ${LIBPNG_LIBS}
 DEPS=config.h 
-OBJ=xsnip.c
+OBJ=xsnip.o
 
 prefix=/usr/local
 exec_prefix=${prefix}
@@ -11,7 +18,7 @@ bindir=${exec_prefix}/bin
 	$(CC) -c -o $@ $< $(CFLAGS)
 
 xsnip: $(OBJ)
-	$(CC) -o $@ $^ $(CFLAGS)
+	$(CC) -o $@ $^ $(LIBS)
 
 install:
 	cp xsnip $(bindir)

--- a/xsnip.c
+++ b/xsnip.c
@@ -156,7 +156,7 @@ int main(int argc, char** argv) {
 
 	display = XOpenDisplay(NULL);
 	if(!display) 
-		exit_clean("Failed to open X display\n");
+		return exit_clean("Failed to open X display\n");
 	root = DefaultRootWindow(display);
 
 	// Query windows for hover selection. This may result in undefined

